spec/tom_queue/deferred_work/deferred_work_set_spec.rb

```diff
@@ -79,16 +79,6 @@
       set.pop(10).should == "work"
     end
 
-    it "should return immediately if it is interrupted by an external thread" do
-      Thread.new { sleep 0.1; set.interrupt }
-      start_time = Time.now
-      set.schedule(start_time + 1.5, "work")
-      set.schedule(start_time + 5, "work")
-      set.pop(10)
-      Time.now.should > start_time + 0.1
-      Time.now.should < start_time + 0.2
-    end
-
```

Comment:
Is it possible to write a test for SignalException handling to make sure we don't regress the behaviour in future?